public/pages/DetectorsList/List/List.tsx

```diff
@@ -125,52 +171,139 @@ export const DetectorList = (props: ListProps) => {
       },
     });
   };
+
+  // Refresh data if user is typing in the index filter
+  const handleSearchIndexChange = debounce(async (searchValue: string) => {
+    if (searchValue !== indexQuery) {
+      const sanitizedQuery = sanitizeSearchText(searchValue);
+      setIndexQuery(sanitizedQuery);
+      await dispatch(getPrioritizedIndices(sanitizedQuery));
+    }
+  }, 300);
+
+  // Refresh data if user is selecting a detector state filter
+  const handleDetectorStateChange = (
+    options: EuiComboBoxOptionProps[]
+  ): void => {
+    let states: DETECTOR_STATE[];
+    states =
+      options.length == 0
+        ? ALL_DETECTOR_STATES
+        : options.map(option => option.label as DETECTOR_STATE);
+    setState(state => ({
+      ...state,
+      selectedDetectorStates: states,
+    }));
+  };
+
+  // Refresh data if user is selecting an index filter
+  const handleIndexChange = (options: EuiComboBoxOptionProps[]): void => {
+    let indices: string[];
+    indices =
+      options.length == 0
+        ? ALL_INDICES
+        : options.map(option => option.label).slice(0, MAX_SELECTED_INDICES);
+
+    setState({
+      ...state,
+      selectedIndices: indices,
+    });
+  };
+
   const handleResetFilter = () => {
     setState(state => ({
       ...state,
       queryParams: {
         ...state.queryParams,
         search: '',
+        indices: '',
       },
+      selectedDetectorStates: ALL_DETECTOR_STATES,
+      selectedIndices: ALL_INDICES,
     }));
   };
 
+  // get all selected detectors
+  const selectedDetectors = filterAndSortDetectors(
+    Object.values(allDetectors),
+    state.queryParams.search,
+    state.selectedIndices,
+    state.selectedDetectorStates,
+    state.queryParams.sortField,
+    state.queryParams.sortDirection,
+    state.queryParams.size,
+    state.page
+  );
```

Comment:
np: Should we just past entire `queryparmas` object?

Reply:
Good catch, yeah. Fixed